examples/solvers/chambolle_pock_denoising.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Denoising using Chambolle-Pock solver."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+# External
+import numpy as np
+import scipy
+import scipy.ndimage
+import matplotlib.pyplot as plt
+
+# Internal
+import odl
+from odl.solvers import chambolle_pock_solver, f_cc_prox_l2_tv, g_prox_none
+
+
+# Read the image
+im = np.rot90(scipy.misc.lena()[::2, ::2], 3)
+shape = im.shape
+# Rescale max to 1
+im = im / np.max(im[:])
+
+# Discretized spaces
+discr_space = odl.uniform_discr([0, 0], shape, shape)
+
+# Original image
+orig = discr_space.element(im)
+
+# Add noise
+im = im + np.random.normal(0, 0.05, shape)
+# Rescale back to [0, 1]
+im = (im - np.min(im)) / (np.max(im) - np.min(im))
+
+# Data: noisy image
+noisy = discr_space.element(im)
+
+# Gradient operator
+grad = odl.DiscreteGradient(discr_space, method='forward')
+
+# Matrix of operators
+K = odl.ProductSpaceOperator([[odl.IdentityOperator(discr_space)], [grad]])
```

Comment:
@kohr-h I had not seen your comment when I was writing mine :)

You might have that the `chambolle_pock_solver` takes operators as input and then internally creates the `ProductSpaceOperator`. However there are a few things to think about.
* The code for the actual solver goes from rather clean to be more involved. Right now it is, more or less, the exact lines given in the algorithmic description.
* In this example there are 2 "parts" to the operator. In general you might want to support and operator with 1 or more parts. For example if you want to run `min | grad x |_1 s.t. Ax = b` you can do that with `G(x) = indicator_function_{Ax = b}`. Or if you for some reason have 3 operators in your objective. But maybe this could be handled dynamically?